lxc/image.go

```diff
@@ -1208,6 +1211,11 @@ func (c *cmdImageList) imageShouldShow(filters []string, state *api.Image) bool
 					}
 				}
 			}
+
+			val, ok := m[key]
+			if ok && val == value {
```

Comment:
value is a string, whereras m contains any elements. 
Would it be better to have structToMap unmarshal into map[string]string?

Reply:
How does string matching against an any that is a bool work later?

Reply:
I just tested it, and it doesn't work. So, a filter like `cached=true` (as `Cached` is a bool in the struct) won't work at the moment. We'd need some logic for this to work.

Reply:
I don't believe we need a lot of logic here, as users will probably only filter by `type` or `architecture`, not by something like `cached` (which is always `false` for `images:` anyway) or `size`.